llama_stack/apis/credentials/credentials.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from datetime import datetime
+from enum import Enum
+from typing import Protocol, runtime_checkable
+
+from pydantic import BaseModel, Field
+
+from llama_stack.schema_utils import webmethod
+
+
+class CredentialTokenType(str, Enum):
+    """The type of credential token.
+
+    :cvar oauth2_authorization_code: OAuth2 authorization code. Used to exchange for an access token (and optionally, refresh token).
+                                     This should be provided once the client receives the OAuth2 callback.
+    :cvar access_token: An opaque OAuth2 / JWT access token. Often what is vended as a "API key".
+
+    """
+
+    oauth2_authorization_code = "oauth2_authorization_code"
```

Comment:
@grs see the new types. we will always use Bearer auth. you either provide an opaque token which will be used as-is, or you provide an authorization code along with a nonce. The nonce is used to look up state and exchange the code for the access token. This assumes that the Stack initiated the authorization request and sent back a 401 beforehand with the nonce.